### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by NymanLauri**

### CODE REVIEW

File: test/locked_restart.jl [hunk 1]
```diff
@@ -18,35 +18,37 @@ end
...
-    A = matrix_with_three_clusters(100)
-    ε = 1e-6
+    for T in (Float64, Complex128)
+        A = matrix_with_three_clusters(T, 100)
+        ε = 1e-6
 
-    # Get the Arnoldi relation after seven restarts.
-    partial_schur = restarted_arnoldi(A, min, max, min, eps(Float64), 4)
+        # Get the Arnoldi relation after seven restarts.
+        partial_schur = restarted_arnoldi(A, min, max, min, eps(T), 4)
```
Comment (haampie) [inline test/locked_restart.jl]: eps(real(T))

**Comment by haampie** [comment]
have you checked whether the tests succeed for the second cluster as well? tests are commented out because it didn't work earlier.

**Comment by haampie** [comment]
From Travis:

```
Implicit restart |    6      6
active = 1
active = 1
active = 4
active = 4
active = 1
active = 1
active = 4
active = 4
```

So that issue remains unresolved up to now.

File: src/implicit_restart.jl [hunk 2]
```diff
@@ -156,13 +156,16 @@ function double_shift!(H_whole::AbstractMatrix{Tv}, min, max, μ::Complex, Q::Ab
         mul!(Q, G₂)
     end
 
-    # Do the last Given's rotation by hand.
-    H[n - 1, n - 2] = H[n + 1, n - 2]
+    if n > 2
```
Comment (haampie) [inline src/implicit_restart.jl]: Why is this necessary btw?

Comment (NymanLauri) [reply src/implicit_restart.jl]: If the active part is only 2 elements wide, then we'd be indexing at [1,0] or [3,0] (it errored once because of this). I'm not sure of just adding "if n > 2" does the exact right thing, but it passed the tests

**Comment by NymanLauri** [comment]
Yeah the tests pass for the second cluster as well. 

The reason for this:
```
Implicit restart |    6      6
active = 1
active = 1
active = 4
active = 4
active = 1
active = 1
active = 4
active = 4 ```

is just that it does the same thing twice, first for real arithmetic, then for complex.

File: src/implicit_restart.jl (hunk 2 above)
Comment (haampie) [reply src/implicit_restart.jl]: Right, it makes sense. Maybe it also makes sense to enforce |max - min| > something.
